src/main/java/org/folio/linked/data/util/BibframeUtils.java

```diff
@@ -118,4 +124,17 @@ public static Resource ensureActive(Resource resource) {
       .orElse(resource);
   }
 
+  public static Resource fetchResource(Identifiable identifiable, ResourceRepository resourceRepository) {
+    Function<String, Optional<Resource>> fetchByIdFunction = id -> resourceRepository.findById(Long.parseLong(id));
+    Function<String, Optional<Resource>> fetchBySrsIdFunction = resourceRepository::findByFolioMetadataSrsId;
+    return Optional.ofNullable(identifiable.getId())
+      .flatMap(fetchByIdFunction)
+      .or(() -> Optional.ofNullable(identifiable.getSrsId())
+        .flatMap(fetchBySrsIdFunction))
+      .map(BibframeUtils::ensureActive)
```

Comment:
We could move it to line 132 to do this operation only when fetching by id. Otherwise fetching by srsId always returns active resource anyway

Reply:
done